src/foam/dao/AbstractDAO.js

```diff
@@ -156,6 +156,16 @@ return new SkipDAO(this.getX(), count, this);
       `,
     },
 
+    {
+      /**
+        Returns an adapted sink by adding no existent sink as well as ad hoc sinks
+      */
+      name: 'toSink',
```

Comment:
I think this is the same thing as prepareSink_(), so we can just remove this method and call prepareSink_() directly from select_().